src/main/java/org/takes/facets/auth/codecs/CcAES.java

```diff
@@ -0,0 +1,167 @@
+     */
+    private final transient byte[] ivbytes;
+
+    /**
+     * The AES secret key object.
+     */
+    private final transient SecretKey secret;
+
+    /**
+     * Ctor.
+     * @param codec Original codec
+     * @param key The encryption key
+     */
+    public CcAES(final Codec codec, final byte[] key) {
+        if (key.length != BLOCK) {
+            throw new IllegalArgumentException(
+                    "Key length must be in 128-bit."
+            );
+        }
+        this.origin = codec;
+        final byte[] passcode = new byte[key.length];
+        System.arraycopy(key, 0, passcode, 0, key.length);
+        final SecureRandom random = new SecureRandom();
+        this.ivbytes = new byte[BLOCK];
+        random.nextBytes(this.ivbytes);
+        this.secret = new SecretKeySpec(passcode, "AES");
+    }
+
+    @Override
+    public byte[] encode(final Identity identity) throws IOException {
+        return this.encrypt(this.origin.encode(identity));
+    }
+
+    @Override
+    public Identity decode(final byte[] bytes) throws IOException {
+        return this.origin.decode(this.decrypt(bytes));
+    }
+
+    /**
+     * Encrypt the given bytes using AES.
+     *
+     * @param bytes Bytes to encrypt
+     * @return Encrypted byte using AES algorithm
+     * @throws IOException for all unexpected exceptions
+     */
+    private byte[] encrypt(final byte[] bytes) throws IOException {
+        try {
+            final Cipher cipher = Cipher.getInstance(ALGORITHM);
+            final AlgorithmParameterSpec spec =
+                    new IvParameterSpec(this.ivbytes);
+            cipher.init(Cipher.ENCRYPT_MODE, this.secret, spec);
+            return cipher.doFinal(bytes);
+        } catch (final InvalidKeyException ike) {
+            throw new IOException(ike);
+        } catch (final NoSuchAlgorithmException nse) {
+            throw new IOException(nse);
+        } catch (final NoSuchPaddingException nspe) {
+            throw new IOException(nspe);
+        } catch (final InvalidAlgorithmParameterException iape) {
+            throw new IOException(iape);
+        } catch (final IllegalBlockSizeException ibse) {
+            throw new IOException(ibse);
+        } catch (final BadPaddingException bpe) {
+            throw new IOException(bpe);
+        }
+    }
+
+    /**
+     * Decrypt the given bytes using AES.
+     *
+     * @param bytes Bytes to decrypt
+     * @return Decrypted bytes
+     * @throws IOException for all unexpected exceptions
+     */
+    private byte[] decrypt(final byte[] bytes) throws IOException {
+        try {
+            final Cipher cipher = Cipher.getInstance(ALGORITHM);
+            final AlgorithmParameterSpec spec =
+                    new IvParameterSpec(this.ivbytes);
+            cipher.init(Cipher.DECRYPT_MODE, this.secret, spec);
+            return cipher.doFinal(bytes);
+        } catch (final InvalidKeyException ike) {
+            throw new IOException(ike);
```

Comment:
@super132 I see, think when we move some code to Ctor this list should shrink a bit